apps/server/src/modules/user-migration/service/user-migration.service.spec.ts

```diff
@@ -0,0 +1,144 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { createMock, DeepMocked } from '@golevelup/ts-jest';
+import { MikroORM } from '@mikro-orm/core';
+import { setupEntities } from '@shared/testing';
+import { BadRequestException, Injectable } from '@nestjs/common';
+import { EntityNotFoundError } from '@shared/common';
+import { SystemService } from '@src/modules/system/service';
+import { SystemDto } from '@src/modules/system/service/dto/system.dto';
+import { OauthConfigDto } from '@src/modules/system/service/dto/oauth-config.dto';
+import { UserMigrationService } from './user-migration.service';
+import { PageContentDto } from './dto/page-content.dto';
+import { PageTypes } from '../interface/page-types.enum';
+
+@Injectable()
+export class UserMigrationServiceSpy extends UserMigrationService {
```

Comment:
why?

Reply:
to be able to access protected methods. in this case the builder for Oauth-Login. It#s not desireable to spy or mock this function, but needed for a proper evaluation. It is not ideal, but better than contains.

Reply:
You should not try to mock your private functions that you are trying to test